core/arch/arm/plat-synquacer/timer_fiq.c

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Linaro Limited
+ */
+
+#include <arm.h>
+#include <console.h>
+#include <drivers/gic.h>
+#include <io.h>
+#include <kernel/panic.h>
+#include <kernel/misc.h>
+#include <rng_pta.h>
+
+static inline uint64_t read_cntps_ctl_el1(void)
+{
+	uint64_t v;
+
+	asm volatile ("mrs %0, cntps_ctl_el1" : "=r" (v));
+
+	return v;
+}
+
+static inline void write_cntps_ctl_el1(uint64_t v)
+{
+	asm volatile ("msr cntps_ctl_el1, %0" : : "r" (v));
+}
+
+static inline uint64_t read_cntps_cval_el1(void)
+{
+	uint64_t v;
+
+	asm volatile ("mrs %0, cntps_cval_el1" : "=r" (v));
+
+	return v;
+}
+
+static inline void write_cntps_cval_el1(uint64_t v)
+{
+	asm volatile ("msr cntps_cval_el1, %0" : : "r" (v));
+}
+
+void generic_timer_start(void)
```

Comment:
It seems that we need a more advanced generic timer driver since we're using them for other purposes as well.

Reply:
Can you please share some details regarding other purposes? So we can add more functionality. Or we can start with basic one and then adding functionality as required.

Reply:
Grep for `cntpct` and you'll see `core/arch/arm/kernel/delay.c` and `core/arch/arm/kernel/tee_time_arm_cntpct.c` doing something with it.

Reply:
Instances you are pointing out uses `cntpct` as a system counter to get ticks. But I have used `cntps_el1` as secure physical timer event/interrupt source which uses `cntpct` as a reference.

Reply:
So there they will not be affected by this then?

Reply:
yes. Also `cntpct` is RO register.